pkg/services/publicdashboards/database/database.go

```diff
@@ -149,118 +152,127 @@ func (d *PublicDashboardStoreImpl) FindByDashboardUid(ctx context.Context, orgId
 		return nil, nil
 	}
 
-	return publicDashboard, err
+	return publicDashboard, nil
 }
 
-// Save Persists public dashboard
-func (d *PublicDashboardStoreImpl) Save(ctx context.Context, cmd SavePublicDashboardCommand) error {
-	if cmd.PublicDashboard.DashboardUid == "" {
-		return dashboards.ErrDashboardIdentifierNotSet
-	}
+// ExistsEnabledByDashboardUid Responds true if there is an enabled public dashboard for a dashboard uid
+func (d *PublicDashboardStoreImpl) ExistsEnabledByDashboardUid(ctx context.Context, dashboardUid string) (bool, error) {
+	hasPublicDashboard := false
+	err := d.sqlStore.WithDbSession(ctx, func(dbSession *db.Session) error {
+		sql := "SELECT COUNT(*) FROM dashboard_public WHERE dashboard_uid=? AND is_enabled=true"
 
-	err := d.sqlStore.WithDbSession(ctx, func(sess *db.Session) error {
-		_, err := sess.UseBool("is_enabled").Insert(&cmd.PublicDashboard)
+		result, err := dbSession.SQL(sql, dashboardUid).Count()
 		if err != nil {
 			return err
 		}
 
-		return nil
+		hasPublicDashboard = result > 0
+		return err
 	})
 
-	return err
+	return hasPublicDashboard, err
 }
 
-// Update updates existing public dashboard
-func (d *PublicDashboardStoreImpl) Update(ctx context.Context, cmd SavePublicDashboardCommand) error {
-	err := d.sqlStore.WithDbSession(ctx, func(sess *db.Session) error {
-		timeSettingsJSON, err := json.Marshal(cmd.PublicDashboard.TimeSettings)
+// ExistsEnabledByAccessToken Responds true if the accessToken exists and the public dashboard is enabled
+func (d *PublicDashboardStoreImpl) ExistsEnabledByAccessToken(ctx context.Context, accessToken string) (bool, error) {
+	hasPublicDashboard := false
+	err := d.sqlStore.WithDbSession(ctx, func(dbSession *db.Session) error {
+		sql := "SELECT COUNT(*) FROM dashboard_public WHERE access_token=? AND is_enabled=true"
+
+		result, err := dbSession.SQL(sql, accessToken).Count()
```

Comment:
Let's do it in this way, as we do for the rest of the db calls. Also change dbSession for sess to be consistent
```suggestion
		publicDashboard := &PublicDashboard{DashboardUid: dashboardUid, IsEnabled: true}
result, err = sess.Count(publicDashboard)
```

Reply:
Inline code change typo (dashboardUid vs accessToken) 🙃

Reply:
OH! I remember. We had to do something silly for xorm to respect boolean types... so we opted not to use this form.